Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx

```diff
@@ -172,106 +170,86 @@ void Clusterer::updateChip(UInt_t ip)
 }
 
 //__________________________________________________
-void Clusterer::finishChip(std::vector<Cluster>& clusters, const MCTruth* labelsDig, MCTruth* labelsClus)
+void Clusterer::finishChip(std::vector<Cluster>* fullClus, std::vector<CompClusterExt>* compClus,
+                           const MCTruth* labelsDig, MCTruth* labelsClus)
 {
   constexpr Float_t SigmaX2 = Segmentation::PitchRow * Segmentation::PitchRow / 12.; // FIXME
   constexpr Float_t SigmaY2 = Segmentation::PitchCol * Segmentation::PitchCol / 12.; // FIXME
 
   const auto& pixData = mChipData->getData();
-  Int_t noc = clusters.size();
+  Int_t noc = compClus ? compClus->size() : (fullClus ? fullClus->size() : 0);
   for (Int_t i1 = 0; i1 < mPreClusterHeads.size(); ++i1) {
     const auto ci = mPreClusterIndices[i1];
     if (ci < 0) {
       continue;
     }
     UShort_t rowMax = 0, rowMin = 65535;
     UShort_t colMax = 0, colMin = 65535;
-    Float_t x = 0., z = 0.;
     int nlab = 0, npix = 0;
     Int_t next = mPreClusterHeads[i1];
     while (next >= 0) {
       const auto& dig = mPixels[next];
       const auto pix = pixData[dig.second];
-      x += pix.getRowDirect();
-      z += pix.getCol();
-      if (pix.getRowDirect() < rowMin) {
-        rowMin = pix.getRowDirect();
-      }
-      if (pix.getRowDirect() > rowMax) {
-        rowMax = pix.getRowDirect();
-      }
-      if (pix.getCol() < colMin) {
-        colMin = pix.getCol();
-      }
-      if (pix.getCol() > colMax) {
-        colMax = pix.getCol();
-      }
       if (npix < mPixArrBuff.size()) {
-        mPixArrBuff[npix] = pix; // needed for cluster topology
-      }
-      if (labelsClus) { // the MCtruth for this pixel is at mChipData->startID+dig.second
-        fetchMCLabels(dig.second + mChipData->getStartID(), labelsDig, nlab);
-      }
-      npix++;
-      next = dig.first;
-    }
-    mPreClusterIndices[i1] = -1;
-    for (Int_t i2 = i1 + 1; i2 < mPreClusterHeads.size(); ++i2) {
-      if (mPreClusterIndices[i2] != ci) {
-        continue;
-      }
-      next = mPreClusterHeads[i2];
-      while (next >= 0) {
-        const auto& dig = mPixels[next];
-        const auto pix = pixData[dig.second]; // PixelData
-        x += pix.getRowDirect();
-        z += pix.getCol();
+        mPixArrBuff[npix++] = pix; // needed for cluster topology
         if (pix.getRowDirect() < rowMin) {
           rowMin = pix.getRowDirect();
-        }
-        if (pix.getRowDirect() > rowMax) {
+        } else if (pix.getRowDirect() > rowMax) {
```

Comment:
I believe, this change should be undone. The checks with respect to rowMin and rowMax must be fully independent.